File: task3.cpp
#include <iostream>
#include <numeric>
#include <cmath>
#include <cstdint>

using namespace std;

uint64_t NOD(uint64_t a, uint64_t b) {
    while (b) { a %= b; swap(a, b); }
    return a;
}

void amount() {
    cout << "Введите а и b: ";
    int a, b;
    if (!(cin >> a >> b)) return;

    if (b == 1) {
        cout << "infinity" << endl;
        return;
    }

    uint64_t A[11][11] = {0};
    for (int n = 1; n <= a; n++) {
        A[n][0] = 1;
        for (int k = 1; k < n; k++) {
            A[n][k] = (k + 1) * A[n - 1][k] + (n - k) * A[n - 1][k - 1];
        }
    }

    uint64_t num = 0;
    for (int k = 0; k < a; k++) {
        num += A[a][k] * (uint64_t)pow(b, a - k);
    }

    uint64_t den = pow(b - 1, a + 1);

    uint64_t common = NOD(num, den);
    cout << num / common << "/" << den / common << endl;
}

int main() {
    amount();
    return 0;
}

File: goFolder/task3/task3.go
package main

import (
	"fmt"
	"math"
)

func NOD(a, b uint64) uint64 {
	for b != 0 {
		a %= b
		a, b = b, a
	}
	return a
}

func main() {
	fmt.Print("Введите a и b: ")
	var a, b int
	fmt.Scan(&a, &b)

	if b == 1 {
		fmt.Println("infinity")
		return
	}

	A := make([][]uint64, a+1)
	for i := range A {
		A[i] = make([]uint64, a+1)
	}

	for n := 1; n <= a; n++ {
		A[n][0] = 1
		for k := 1; k < n; k++ {
			A[n][k] = uint64(k+1)*A[n-1][k] + uint64(n-k)*A[n-1][k-1]
		}
	}

	var num uint64 = 0
	for k := 0; k < a; k++ {
		num += A[a][k] * uint64(math.Pow(float64(b), float64(a-k)))
	}

	den := uint64(math.Pow(float64(b-1), float64(a+1)))

	common := NOD(num, den)
	fmt.Printf("%d/%d\n", num/common, den/common)
}
